lib/java_buildpack/framework/sealights_agent.rb

```diff
@@ -0,0 +1,168 @@
+# frozen_string_literal: true
+
+# Cloud Foundry Java Buildpack
+# Copyright 2013-2020 the original author or authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'java_buildpack/component/versioned_dependency_component'
+require 'java_buildpack/framework'
+require 'uri'
+require 'net/http'
+require 'pathname'
+require 'base64'
+require 'json'
+
+module JavaBuildpack
+  module Framework
+
+    # Encapsulates the functionality for enabling zero-touch Sealights support.
+    class SealightsAgent < JavaBuildpack::Component::BaseComponent
+
+      def initialize(context)
+        super(context)
+        @logger = JavaBuildpack::Logging::LoggerFactory.instance.get_logger SealightsAgent
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#detect)
+      def detect
+        supports? ? "#{self.class.to_s.dash_case}=latest" : nil
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        announce("'compile' method is starting")
```

Comment:
There's quite a bit of debug logging like this entry here. Please remove that for your final code submission.